app/models/manageiq/providers/infra_manager/provision_workflow.rb

```diff
@@ -5,7 +5,8 @@ def set_or_default_hardware_field_values(vm)
       :cpu_limit      => vm.cpu_limit,
       :memory_limit   => vm.memory_limit,
       :cpu_reserve    => vm.cpu_reserve,
-      :memory_reserve => vm.memory_reserve
+      :memory_reserve => vm.memory_reserve,
+      :disk_size      => vm.hardware.disks.find_by(:device_type => "disk").size / (1024**3)
```

Comment:
A few things.

In [hardware.rb](https://github.com/ManageIQ/manageiq/blob/master/app/models/hardware.rb#L12-L15) I notice that we have a number of different `device_type` values called out but none are `"disk"` in particular. I can't imagine that this is a new concept. Maybe we want to use `hard_disks` instead?

`size` is the count of the disks (e.g.: 1-4), and ` / (1024**3)` is definitely talking about disk storage (e.g.: 3.5 * 10^9), so there is a disconnect here.

I'm thinking `allocated_disk_storage` or `used_disk_storage` would work here. The desire to not display CDRom values in disk size makes me wonder if `used_disk_storage` should not `disks.sum(:used_disk_storage)` but `hard_disks.sum(:used_disk_storage)`. (I'm not sure if we've tested a virtual sum with a `where()` clause, so we probably want to put a test in here and over in `virtual_attributes` gem.

Reply:
@kbrock changed to `allocated_disk_storage`

Reply:
@kbrock I think allocated_disk_storage is confusing here, this is the size of a single disk that we're going to be creating and isn't intended to map to any columns.